File: backend/app/schemas/document.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.core.constants import (
    DOCUMENT_STATUS_UPLOADED,
    DOCUMENT_TYPE_UNKNOWN,
    DOCUMENT_UPLOAD_SOURCE_MANUAL,
)


class DocumentBase(BaseModel):
    workspace_id: UUID
    original_filename: str
    stored_filename: str
    file_extension: str
    content_type: str | None = None
    file_size_bytes: int
    storage_backend: str
    storage_bucket: str | None = None
    storage_key: str
    document_type: str = DOCUMENT_TYPE_UNKNOWN
    classification_confidence: float | None = None
    classification_method: str | None = None
    classification_reason: dict[str, Any] | None = None
    classified_at: datetime | None = None
    status: str = DOCUMENT_STATUS_UPLOADED
    upload_source: str = DOCUMENT_UPLOAD_SOURCE_MANUAL


class DocumentCreate(DocumentBase):
    id: UUID


class DocumentResponse(DocumentBase):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    updated_at: datetime


class DocumentDownloadUrlResponse(BaseModel):
    document_id: UUID
    download_url: str
    expires_in_seconds: int

File: frontend/src/types/document.ts
export type DocumentStatus = 'uploaded' | 'processing' | 'processed' | 'failed' | 'deleted'

export interface Document {
  id: string
  workspace_id: string
  original_filename: string
  stored_filename: string
  file_extension: string
  content_type?: string | null
  file_size_bytes: number
  storage_backend: string
  document_type: string
  classification_confidence?: number | null
  status: DocumentStatus
  created_at: string
  updated_at: string
}

export interface ProcessingJobSummary {
  id: string
  status: string
  job_type: string
}

export interface UploadDocumentResponse {
  document: Document
  processing_job: ProcessingJobSummary
}

export interface DocumentDownloadUrlResponse {
  document_id: string
  download_url: string
  expires_in_seconds: number
}
